### PR TITLE
[NO_TICKET] Reorganize exports

### PR DESCRIPTION
**Pull request by bernardwang**
## Summary
- Add `index.js` to each component folder
- Simplify exports in the root entry `index.js`
- Factor out `getMonthNames` and `maskHelpers` into separate files

## Changes
- Export `maskValue`

## How to test
- Ensure all components are still avaliable
- Run `yarn build` to see no changes in the resulting bundle

### CODE REVIEW

**Review by line47 — APPROVED** [review]
LGTM!

**Comment by line47** [comment]
@bernardwang updated FormControl to fix sort-import issue

File: packages/design-system/src/components/FormControl/FormControl.tsx [hunk 1]
```diff
@@ -2,8 +2,8 @@ import FormLabel from '../FormLabel/FormLabel';
 import InlineError from '../InlineError/InlineError';
 import React from 'react';
 import classNames from 'classnames';
-import uniqueId from 'lodash.uniqueid';
 import { errorPlacementDefault } from '../flags';
+import uniqueId from 'lodash.uniqueid';
```
Comment (bernardwang) [inline packages/design-system/src/components/FormControl/FormControl.tsx:6]: thanks!

**Review by nichia — APPROVED** [review]
LGTM

**Pull request merged**